src/breeze/components/breeze/sidebar_link.cr

```diff
@@ -0,0 +1,26 @@
+class Breeze::SidebarLink(T) < BreezeComponent
+  needs context : HTTP::Server::Context
+  needs content : String
+
+  def render
+    link content, to: T, class: link_class(T)
```

Comment:
Rather than using `T` directly you could add a `needs` that would allow people to link to things other than an action class:

```crystal
needs link_to : T

def render
  link to: link_to, class: link_class(T)
end
```

I may be missing stuff though so if this doesn't work, feel free to ignore it!

Reply:
I originally wanted to do that, but I was getting an error similar to this https://github.com/luckyframework/lucky/issues/1243 For some reason, when using dynamic Actions, the compiler freaks out.

Reply:
Ok no prob! I think what you have is good to start! The only other possibility I see is passing it a proc instead. Like we do with DescriptionList and stuff. So you can render a link however you want. But I think that is probably overkill

Reply:
I'm giving this another shot, but still haven't figured out how to get around that weird issue:

```crystal
  Habitat.create do
    setting links : Array(Proc(Breeze::SidebarLinks, Nil)) = [
      ->(breeze : Breeze::SidebarLinks) {
        breeze.mount(Breeze::SidebarLink, context: breeze.context, link_text: "Requests", link_to: Breeze::Requests::Index)
      }
    ]
  end

  def render
    settings.links.each do |component|
      component.call(self)
    end
  end
```

gives me this error:

```
web          |  > 3 |
web          |  > 4 |
web          |  > 5 |     add_route(:put, "/projects/:project_id", Projects::Update)
web          |            ^
web          | Error: Invalid call to Projects::Update.route
web          |
web          | Required arguments:
web          |
web          | - project_id
web          |
```

because that `Breeze::SidebarLink` component calls `link link_text, to: link_to, class: link_class(link_to)`. The dynamic `link_to` somehow thinks it needs to work with every possible combination of Lucky Action or something 😕

Reply:
ok, I got this working... I'm just passing in a raw String now. Not as type-safe as I'd like, but I couldn't find any other way around it.